desktop/main.dev.ts

```diff
@@ -61,6 +62,7 @@ let browserView: BrowserView;
 let tray: Tray;
 let nodeManager: NodeManager;
 let notificationManager: NotificationManager;
+let updateManager: UpdateManager;
```

Comment:
Solves the problem of it using the UpdateManager before initialization. 
I'll move the initialization to before the `addIpcEventListeners()`

Reply:
> Solves the problem of it using the UpdateManager before initialization.
Yep, but I hope you understand that it solves the error `updateManager is not defined`, but the `updateManager` still can be equal to `undefined` and get another error: `Cannot read property 'checkForUpdates' of undefined`. It is not occurs (I hope) only because it called by IPC event and it happens after the initialization of `UpdateManager`.

What I'm trying to say — much better to avoid writing code patterns that lead us to keep in mind the order of calls, async calls, mutations, and so on. Even we don't use the FP :)

1. So, `UpdateManager` is created within `createWindow` function, so it can be initialized there without any references in the global namespace, which also can be `undefined`.
2. Then you can pass some dependencies inside `addIpcEventListeners`. But since we have a class that handles App updating, I propose to add a subscription on events right in the `UpdateManager` class.

Finally, you'll end up with two changed lines of code in the `main.dev.ts` -> one for import UpdateManager, and the second one to construct the instance.

Reply:
TBH, I feel comfortable making the **updateManager** `undefined` and initialize it before use. Passing props to the `addIpcEventListeners` is a bit of overkill for me.

Reply:
Fixed